src/main/kotlin/org/rust/ide/formatter/RustBlock.kt

```diff
@@ -21,6 +26,23 @@ class RustBlock(
     }
 
     override fun getSpacing(child1: Block?, child2: Block): Spacing? {
+        if (child1 is AbstractRustBlock && child2 is AbstractRustBlock) {
+            val elt1 = child1.node.elementType
+            val elt2 = child2.node.elementType
+            val psi1 = child1.node.psi
+            val psi2 = child2.node.psi
+            val parent = psi1.parent
+
+            if (psi1 is RustOuterAttr && (psi2 is RustOuterAttr || parent is RustItem)) {
+                return RustSpacing.lineBreak()
+            }
+
+            if (psi1.isTopLevelItem && psi2.isTopLevelItem) {
+                return RustSpacing.lineBreak(
+                    keepLineBreaks = ctx.commonSettings.KEEP_LINE_BREAKS,
+                    keepBlankLines = ctx.commonSettings.KEEP_BLANK_LINES_IN_DECLARATIONS)
```

Comment:
This rule uses IDEA's default which states +/-: "keep blank lines, but maximally 2 consecutive"